test/rules/curly.js

```diff
@@ -0,0 +1,54 @@
+module.exports = {
+	valid: [
+		"if (foo) {\n" +
+		"\t\tfoo++;\n" +
+		"\t}",
+		"while (bar) {\n" +
+		"\t\tbaz();\n" +
+		"\t}",
+		"if (foo) {\n" +
+		"\t\tbaz();\n" +
+		"\t} else {\n" +
+		"\t\tqux();\n" +
+		"}",
+		"switch (false) {\n" +
+		"    case true: {\n" +
+		"      let x = \"bar\";\n" +
+		"      console.log(x);\n" +
+		"      break;\n" +
+		"    }\n" +
+		"\n" +
+		"    case false: {\n" +
+		"      let x = \"baz\";\n" +
+		"      console.log(x);\n" +
+		"      break;\n" +
+		"    }\n" +
+		"}",
+	],
+	invalid: [
+		{
+			code: "if (foo) foo++;",
+			errors: ["Expected { after \'if\' condition."]
```

Comment:
Within a double quoted string, single quotes don't need to be escaped. And vice versa.